### PR TITLE
[assets] font sizes consolidation

### PR DESCRIPTION
**Pull request by clairesunstudio**
Issue: font sizes in mayflower is all over the place, lacking a consistent scale
Screen Shot 2020-05-13 at 9 27 32 AM

Goal: create a consistent and flexible font system for switching out Texta and for future maintainability.

In order to consolidate the font size variants, we need to：

- come up with a scale system that makes sense for the current usage
- consolidate all units into one (em, px --> rem)
- standardize all existing font sizes into using the new scale and unit
 

To see the impact of this change, I have taken the following steps to isolate visual regressions:

1. do the exact mapping so there should be no visual regression
2. standardize the rest of rem values to the closest values in the variable scale 
3. convert `px` to `rem` --> https://circleci.com/gh/massgov/mayflower/27003?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
4. convert `em` to `rem`--> section links are slightest bigger to use `$fonts-xlarge` see 28b9756
 

After:
All font sizes will reference the variables going forward
```
$fonts-base:        16px;
$fonts-base-mobile: 14.5px;
$fonts-print:       12px;
// ----------------------------
$fonts-3xsmall:     0.8rem;
$fonts-2xsmall:     0.875rem;
$fonts-xsmall:      1rem;
$fonts-smaller:     1.125rem;
$fonts-small:       1.25rem;
$fonts-medium:      1.375rem;
$fonts-large:       1.5rem;
$fonts-larger:      1.625rem;
$fonts-xlarge:      1.75rem;
$fonts-2xlarge:     2rem;
$fonts-3xlarge:     2.25rem;
$fonts-4xlarge:     2.5rem;
$fonts-5xlarge:     2.75rem;
$fonts-6xlarge:     3rem;
$fonts-7xlarge:     3.25rem;
$fonts-8xlarge:     3.5rem;
$fonts-9xlarge:     4rem;
$fonts-10xlarge:    5rem;
$fonts-11xlarge:    7rem;
$fonts-max:         12rem;
```

### CODE REVIEW

File: assets/scss/01-atoms/_button-tag.scss [hunk 1]
```diff
@@ -14,7 +14,7 @@
   &:after {
     content: "+";
     display: inline-block;
-    font-size: 1.5em;
+    font-size: 1.5rem;
```
Comment (clairesunstudio) [inline assets/scss/01-atoms/_button-tag.scss]:
```suggestion
    font-size: 1.5em;
```

File: assets/scss/01-atoms/_input-checkbox.scss [hunk 2]
```diff
@@ -32,7 +32,7 @@
       content: "";
       display: inline-block;
       height: 2em;
-      font-size: .5em;
+      font-size: .5rem;
```
Comment (clairesunstudio) [inline assets/scss/01-atoms/_input-checkbox.scss]:
```suggestion
      font-size: .5em;

```

File: assets/scss/01-atoms/_input-group.scss [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
     &:after {
       content: "(optional)";
-      font-size: .8em;
+      font-size: .8rem;
```
Comment (clairesunstudio) [inline assets/scss/01-atoms/_input-group.scss]:
```suggestion
      font-size: .8em;

```

File: assets/scss/01-atoms/_input-radio.scss [hunk 4]
```diff
@@ -21,7 +21,7 @@
       content: "\2022";
       display: inline-block;
       height: .45em;
-      font-size: 2.5em;
+      font-size: 2.5rem;
```
Comment (clairesunstudio) [inline assets/scss/01-atoms/_input-radio.scss]:
```suggestion
      font-size: 2.5em;

```

File: assets/scss/01-atoms/_label.scss [hunk 5]
```diff
@@ -10,7 +10,7 @@
 
     &:after {
       content: "(optional)";
-      font-size: .8em;
+      font-size: .8rem;
```
Comment (clairesunstudio) [inline assets/scss/01-atoms/_label.scss]:
```suggestion
      font-size: .8em;

```

File: assets/scss/01-atoms/_select-box.scss [hunk 6]
```diff
@@ -11,7 +11,7 @@
 
     &:after {
       content: "(optional)";
-      font-size: .8em;
+      font-size: .8rem;
```
Comment (clairesunstudio) [inline assets/scss/01-atoms/_select-box.scss]:
```suggestion
      font-size: .8em;

```

File: assets/scss/02-molecules/_section-links.scss [hunk 7]
```diff
@@ -118,13 +118,13 @@
     @media ($bp-x-small-max) {
       align-self: flex-start;
       cursor: pointer;
-      font-size: 1.25em;
+      font-size: 1.25rem;
```
Comment (clairesunstudio) [inline assets/scss/02-molecules/_section-links.scss]:
```suggestion
      font-size: $fonts-medium;

```

File: assets/scss/02-molecules/_section-links.scss [hunk 8]
```diff
@@ -118,13 +118,13 @@
...
       cursor: pointer;
-      font-size: 1.25em;
+      font-size: 1.25rem;
       padding-right: 1em;
 
       &:after {
         content: "+";
         display: block;
-        font-size: 1.5em;
+        font-size: 1.5rem;
```
Comment (clairesunstudio) [inline assets/scss/02-molecules/_section-links.scss]:
```suggestion
        font-size: 1.5em;

```

File: assets/scss/02-molecules/_tooltip.scss [hunk 9]
```diff
@@ -79,7 +79,7 @@
     &:after {
       content: "+";
       display: inline-block;
-      font-size: 2.5em;
+      font-size: 2.5rem;
```
Comment (clairesunstudio) [inline assets/scss/02-molecules/_tooltip.scss]:
```suggestion
      font-size: 2.5em;

```

File: assets/scss/03-organisms/_page-banner.scss [hunk 10]
```diff
@@ -296,7 +296,7 @@ $page-banner-height: 330px;
     z-index: 1;
 
     & > abbr {
-      font-size: .4em;
+      font-size: .4rem;
```
Comment (clairesunstudio) [inline assets/scss/03-organisms/_page-banner.scss]:
```suggestion
      font-size: .4em;

```

File: assets/scss/03-organisms/_rich-text.scss [hunk 11]
```diff
@@ -49,7 +49,7 @@ $heading-indent: $gutter;
   & &__footnote {
     border-style: solid;
     border-width: 2px;
-    font-size: .6em;
+    font-size: .6rem;
```
Comment (clairesunstudio) [inline assets/scss/03-organisms/_rich-text.scss]:
```suggestion
    font-size: .6em;

```

File: assets/scss/08-print/_print.scss [hunk 12]
```diff
@@ -65,7 +65,7 @@
 
   a::after {
     content: " ("attr(href)")";
-    font-size: 0.8em;
+    font-size: 0.8rem;
```
Comment (clairesunstudio) [inline assets/scss/08-print/_print.scss]:
```suggestion
    font-size: 0.8em;
```

**Comment by clairesunstudio** [comment]
For text, we should be setting `font-size` with font size variables in _variables.scss. For icon, or content in the pseudo-class, e.g. `:after`, we can use `em` to specify the ratio between its font size and the parent font-size, in this case, the font size of the main class

**Comment by clairesunstudio** [comment]
Mayflower React diffs --> https://27079-137521197-gh.circle-artifacts.com/0/backstop/data/html_report/index.html

**Review by joeg8612 — APPROVED** [review]
This passes a functional review when pulling this into Drupal and doing a backstop check for differences.

File: assets/scss/00-base/_variables.scss [hunk 13]
```diff
@@ -42,9 +43,41 @@ $fonts-vietnamese: "Muli";
...
+$fonts-6xlarge:     3rem !default;      // H1 small
+$fonts-7xlarge:     3.25rem !default;   // H1 medium
+$fonts-8xlarge:     3.5rem !default;    // H1 large
+$fonts-9xlarge:     4rem !default;
+$fonts-10xlarge:    5rem !default;
+$fonts-11xlarge:    7rem !default;
+$fonts-max:         12.75rem !default;  // Error decorative letters
+
+//letter-spacing for Texta
+$letter-spacing-large:   .1em;
```
Comment (smurrayatwork) [inline assets/scss/00-base/_variables.scss]:
```suggestion
$letter-spacing-large:   0.1em;
```

File: assets/scss/00-base/_variables.scss [hunk 14]
```diff
@@ -42,9 +43,41 @@ $fonts-vietnamese: "Muli";
...
+$letter-spacing-medium:  .05em;
```
Comment (smurrayatwork) [inline assets/scss/00-base/_variables.scss]:
```suggestion
$letter-spacing-medium:  0.05em;
```

File: assets/scss/00-base/_variables.scss [hunk 15]
```diff
@@ -42,9 +43,41 @@ $fonts-vietnamese: "Muli";
...
+$letter-spacing-small:   .02em;
```
Comment (smurrayatwork) [inline assets/scss/00-base/_variables.scss]:
```suggestion
$letter-spacing-small:   0.02em;
```

**Review by smurrayatwork — APPROVED** [review]

**Pull request merged**